portal/get_patient_info.php

```diff
@@ -75,18 +75,37 @@
 DEFINE("COL_PID", "pid");
 DEFINE("COL_POR_PWD", "portal_pwd");
 DEFINE("COL_POR_USER", "portal_username");
+DEFINE("COL_POR_LOGINUSER", "portal_login_username");
 DEFINE("COL_POR_SALT", "portal_salt");
 DEFINE("COL_POR_PWD_STAT", "portal_pwd_status");
+
+// This is now an account id. Legacy username. Will force case sensitive regardless of collation
 $sql = "SELECT " . implode(",", array(
-    COL_ID,
-    COL_PID,
-    COL_POR_PWD,
-    COL_POR_SALT,
-    COL_POR_PWD_STAT
-)) . " FROM " . TBL_PAT_ACC_ON . " WHERE " . COL_POR_USER . "=?";
+        COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) . " FROM " . TBL_PAT_ACC_ON .
+    " WHERE BINARY " . COL_POR_USER . "= ?";
 $auth = privQuery($sql, array(
     $_POST['uname']
 ));
+if ($password_update === 2) {
+    $auth[COL_POR_LOGINUSER] = '';
+}
+// real username
+if ($auth !== false && !empty($auth[COL_POR_LOGINUSER])) {
+    $sql = "SELECT " . implode(",", array(
+            COL_ID, COL_PID, COL_POR_PWD, COL_POR_USER, COL_POR_LOGINUSER, COL_POR_SALT, COL_POR_PWD_STAT)) . " FROM " . TBL_PAT_ACC_ON .
+        " WHERE BINARY " . COL_POR_LOGINUSER . "= ?";
```

Comment:
Doesn't this mean we need to ensure that COL_POR_LOGINUSER is now unique for each user?

Reply:
I fixed this.